module_renamer/commands/track_modifications.py

```diff
@@ -0,0 +1,206 @@
+
+
+def write_list_to_file(list_with_modified_imports, file_name):
+    """
+    Write the list of modified imports on a python file, the python file per default will be named
+    "list_output.py" but can be changed by passing the argument --output_file
+
+    The name of the list (inside the file) cannot be changed since it will be used later on the
+    script for renaming the project.
+    """
+    echo('Generating the file {0}'.format(file_name))
+
+    with open(file_name, 'w') as file:
+        file.write("imports_to_move = ")
+        file.writelines(repr(list(list_with_modified_imports)))
+
+    FormatFile(file_name, in_place=True)
+
+
+def generate_list_with_modified_imports(origin_import_list, working_import_list):
+    """
+    This methods looks for imports that keep the same name but has has different modules path
+
+    :return: A list with unique elements that has the same name but different modules path
+    :rtype: set
+    """
+
+    origin_filtered, working_filtered = _filter_import(origin_import_list, working_import_list)
+
+    moved_imports = _find_moved_imports(origin_filtered, working_filtered)
+
+    list_with_modified_imports = _check_for_conflicts(moved_imports)
+
+    return list_with_modified_imports
+
+
+def _filter_import(origin_import_list, working_import_list):
+    """
+    Create a new set with elements present origin_list or working_list but not on both,
+    this helps to filter classes that could have same name but different modules.
+    """
+    difference = origin_import_list.symmetric_difference(working_import_list)
+    origin_filtered = origin_import_list.intersection(difference)
+    working_filtered = working_import_list.intersection(difference)
+    return origin_filtered, working_filtered
+
+
+def _find_moved_imports(origin_filtered, working_filtered):
+    list_with_modified_imports = {
+        (origin.module + "." + origin.name, working.module + "." + working.name)
+        for origin in origin_filtered
+        for working in working_filtered
+        if working.name is not '*'
+        if origin.name == working.name
+        if origin.module != working.module
+    }
+    return list_with_modified_imports
+
+
+def _check_for_conflicts(list_with_modified_imports):
+    import_from = [modified_import[0] for modified_import in list(list_with_modified_imports)]
+    imports_with_conflict = [class_name
+                             for class_name, number_of_occurrences in Counter(import_from).items()
+                             if number_of_occurrences > 1]
+    if len(imports_with_conflict) > 0:
+        echo(CONFLICT_MSG.format('\n -> '.join(imports_with_conflict)))
+
+        if confirm(INFORMATIVE_CONFLICT_MSG, abort=True):
+            list_with_modified_imports = [modified_import for modified_import in
+                                          list_with_modified_imports
+                                          if modified_import[0] not in imports_with_conflict
```

Comment:
Each one of these `if`s is a linear search, depending on the number of imports this might impact the performance.